File: houseworker/src/main/java/com/example/moodtherapist/ui/theme/TaskAdapter.java
package com.example.moodtherapist.ui.theme;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shared.R;
import com.example.shared.Task;

import java.util.List;

public class TaskAdapter extends RecyclerView.Adapter<TaskAdapter.TaskViewHolder> {

    private final List<Task> taskList;

    public TaskAdapter(List<Task> tasks) {
        this.taskList = tasks;
    }

    public static class TaskViewHolder extends RecyclerView.ViewHolder {
        TextView textView;
        TextView statusView;

        public TaskViewHolder(View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.tvTaskText);
            statusView = itemView.findViewById(R.id.tvTaskStatus);
        }
    }

    @NonNull
    @Override
    public TaskViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_task, parent, false);
        return new TaskViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TaskViewHolder holder, int position) {
        Task task = taskList.get(position);
        holder.textView.setText(task.getText());
        holder.statusView.setText("Statut : " + task.getStatus());
    }

    @Override
    public int getItemCount() {
        return taskList.size();
    }
}

File: shared/src/main/java/com/example/shared/TaskAdapter.kt
package com.example.shared

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.shared.R

class TaskAdapter(
    private val tasks: List<Task>,
    private val showDoneButton: Boolean = false,
    private val onTaskDone: ((Task) -> Unit)? = null
) : RecyclerView.Adapter<TaskAdapter.TaskViewHolder>() {

    class TaskViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val text: TextView = view.findViewById(R.id.tvTaskText)
        val status: TextView = view.findViewById(R.id.tvTaskStatus)
        val btnDone: Button = view.findViewById(R.id.btnDone)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TaskViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_task, parent, false)
        return TaskViewHolder(view)
    }

    override fun onBindViewHolder(holder: TaskViewHolder, position: Int) {
        val task = tasks[position]
        holder.text.text = task.text
        holder.status.text = "Statut : ${task.status}"

        if (showDoneButton && task.status != "done") {
            holder.btnDone.visibility = View.VISIBLE
            holder.btnDone.setOnClickListener {
                onTaskDone?.invoke(task)
            }
        } else {
            holder.btnDone.visibility = View.GONE
        }
    }

    override fun getItemCount(): Int = tasks.size
}
